### PR TITLE
Adding getPostProcessScript parameter.

### PR DESCRIPTION
**Pull request by iclanton**

### CODE REVIEW

File: common/changes/@microsoft/set-webpack-public-path-plugin/ianc-set-public-path-postprocess_2017-08-02-08-01.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/set-webpack-public-path-plugin",
+      "comment": "Added a new getPostProcessScript configuration option to allow consumers to do arbitrary post-processing on the webpack public path variable.",
```
Comment (nickpape-msft) [inline common/changes/@microsoft/set-webpack-public-path-plugin/ianc-set-public-path-postprocess_2017-08-02-08-01.json]: >Added [](start = 18, length = 5)

Add

File: webpack/set-webpack-public-path-plugin/src/codeGenerator.ts [hunk 2]
```diff
@@ -45,54 +47,77 @@ export function getSetPublicPathCode(options: IInternalOptions, emitWarning: (wa
...
       '    }',
       '  }',
       '}'
     ]);
+
+    if (options.getPostProcessScript) {
+      lines.push(...[
+        '',
+        `if (${varName}) {`,
+        `  ${options.getPostProcessScript(varName)};`,
```
Comment (nickpape-msft) [inline webpack/set-webpack-public-path-plugin/src/codeGenerator.ts]: this is kind of weird. we should probably set varname to the return value of this function.

File: webpack/set-webpack-public-path-plugin/src/codeGenerator.ts [hunk 3]
```diff
@@ -45,54 +47,77 @@ export function getSetPublicPathCode(options: IInternalOptions, emitWarning: (wa
...
       lines.push(
         '',
-        `publicPath += '${appendSlashAndEscapeSingleQuotes(options.urlPrefix)}';`);
+        `${varName} += '${appendSlashAndEscapeSingleQuotes(options.urlPrefix)}';`);
     }
 
     lines.push(
       '',
-      `${options.webpackPublicPathVariable} = publicPath;`
+      `${options.webpackPublicPathVariable} = ${varName};`
```
Comment (nickpape-msft) [inline webpack/set-webpack-public-path-plugin/src/codeGenerator.ts]: this bit of code looks repeated in both if/else blocks

File: common/changes/@microsoft/set-webpack-public-path-plugin/ianc-set-public-path-postprocess_2017-08-02-08-01.json (hunk 1 above)
Comment (nickpape-msft) [inline common/changes/@microsoft/set-webpack-public-path-plugin/ianc-set-public-path-postprocess_2017-08-02-08-01.json]: (use the present tense)

---
In reply to: [131004654](https://github.com/Microsoft/web-build-tools/pull/291#discussion_r131004654) [](ancestors = 131004654)

File: webpack/set-webpack-public-path-plugin/src/codeGenerator.ts (hunk 2 above)
Comment (iclanton) [inline webpack/set-webpack-public-path-plugin/src/codeGenerator.ts]: It's not a function, it's a code snippet. This function returns a block of script as text that modifies the variable, not an actual function.

**Comment by nickpape-msft** [comment]
:shipit:

**Comment by nickpape-msft** [comment]
Approved with Suggestions

**Pull request merged**